extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/iam/IamServiceImpl.java

```diff
@@ -172,18 +209,40 @@ private Supplier<IamCredentialsClient> defaultIamCredentialsClientSupplier() {
         }
     }
 
-    private record ApplicationDefaultCredentials(Monitor monitor) implements AccessTokenProvider {
+    private record DefaultCredentialsUtil(Monitor monitor) implements CredentialsUtil {
```

Comment:
please extract to a separate class and document

Reply:
Done